From 48c84bd037128d6788ee74786ac5f4aca9727ae9 Mon Sep 17 00:00:00 2001
From: Antonio Larrosa <alarrosa@suse.com>
Date: Wed, 20 Dec 2023 13:48:14 +0100
Subject: [PATCH] Fix race condition generating docs

When doing a parallel build, there's a race condition
that sometimes builds qtpdf before pdf_attributions is
generated resulting in an incomplete qtpdf.index file.
Adding this dependency solves this.

Issue found by Bernhard Wiedemann <bwiedemann@suse.com> while
testing build reproducibility of packages in openSUSE and
reported originally at https://bugzilla.opensuse.org/show_bug.cgi?id=1217774

Change-Id: Ia7c0c4c685eeed56e9318871467dc2be6626cc47
---
 src/pdf/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/pdf/CMakeLists.txt b/src/pdf/CMakeLists.txt
index a50120d3e..c40476f89 100644
--- a/src/pdf/CMakeLists.txt
+++ b/src/pdf/CMakeLists.txt
@@ -66,6 +66,7 @@ add_code_attributions_target(
 )
 add_dependencies(generate_pdf_attributions run_pdf_GnDone)
 add_dependencies(docs generate_pdf_attributions)
+add_dependencies(prepare_docs_Pdf generate_pdf_attributions)
 
 ##
 #   TOOLCHAIN SETUP
-- 
2.42.0

